File: handlerclaw/lib/features/api-keys/domain/entities/api_key_entity.dart
enum ApiKeyStatus {
  active,
  revoked,
  expired,
}

class ApiKeyEntity {
  final String id;
  final String name;
  final String keyPreview;
  final ApiKeyStatus status;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String? plainKey;

  ApiKeyEntity({
    required this.id,
    required this.name,
    required this.keyPreview,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
    this.plainKey,
  });

  bool get isActive => status == ApiKeyStatus.active;
  bool get isRevoked => status == ApiKeyStatus.revoked;
}

File: api/src/modules/(jwt-auth)/api-keys/domain/entities/api-key.entity.ts
import type { ApiKeyStatus } from "../../../../../lib/generated/prisma/enums.js";

export type IApiKey = {
  id: string;
  user_id: string;
  key_preview: string;
  key_full: string;
  name: string;
  status: ApiKeyStatus;
  created_at: Date;
  updated_at: Date;
  deleted_at: Date | null;
};
